packages/world/src/systems/libraries/TransferLib.sol

```diff
@@ -53,13 +52,38 @@ library TransferLib {
     EntityId chestEntityId = isDeposit ? baseDstEntityId : baseSrcEntityId;
     VoxelCoord memory chestCoord = isDeposit ? dstCoord : srcCoord;
 
+    VoxelCoord memory shardCoord = chestCoord.toLocalEnergyPoolShardCoord();
+    uint128 energyCost = PLAYER_TRANSFER_ENERGY_COST;
+
     EntityId forceFieldEntityId = getForceField(chestCoord);
     uint256 machineEnergyLevel = 0;
     if (forceFieldEntityId.exists()) {
+      energyCost += SMART_CHEST_ENERGY_COST;
       EnergyData memory machineData = updateMachineEnergyLevel(forceFieldEntityId);
       machineEnergyLevel = machineData.energy;
+      require(machineData.energy >= SMART_CHEST_ENERGY_COST, "Machine does not have enough energy");
+      Energy._set(
+        forceFieldEntityId,
+        EnergyData({ energy: machineData.energy - SMART_CHEST_ENERGY_COST, lastUpdatedTime: uint128(block.timestamp) })
+      );
+    }
+
+    {
+      uint128 playerEnergy = Energy._getEnergy(playerEntityId);
```

Comment:
same with this, could become `playerEntityId.decreaseEnergy(amount);`

Reply:
oh and make `PlayerEntity` a user type?

Reply:
Added a separate issue for this https://github.com/tenetxyz/biomes-contracts/issues/42

Reply:
nvm added a util `decreaseEnergy`